Woodoku/src/UI/InterfaceUI.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+from Woodoku.src.Entity.WoodokuBoard import WoodokuBoard
+from Woodoku.src.Entity.WoodokuShape import WoodokuShape
+
+
+class InterfaceUI(ABC):
+
+    @abstractmethod
+    def show_start_game(self, board) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def show_board(self, board: WoodokuBoard) -> None:
+        """
+        Show the board and score to user.
+
+        :param board: The board
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def choose_shape(self, shapes: List[WoodokuShape], shape_availability: List[bool]) -> int:
+        """
+        Precondition: len(shapes) == len(shape_availability)
+
+        Show the UI to let user choose one of the shapes.
+
+        :param shapes: All the shapes in this round, no matter if chosen before.
+        :param shape_availability: If the shape in <shapes> is still waiting to be chosen.
+        :return: The index of shape chosen from user.
```

Comment:
The index of shape chosen by the user

Reply:
Yes, but since this is a UI class, it should implement the logic for the user to choose the index. For example, in the `CommandLineUI`, we could expact that the method `choose_shape` can print prompt to the command line and let user input the index.

Reply:
Just updated!